cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/RemoteTaskExecutor.java

```diff
@@ -80,35 +90,87 @@ public RemoteTaskExecutor(CConfiguration cConf, RemoteClientFactory remoteClient
    * @throws Exception returned by remote task if any
    */
   public byte[] runTask(RunnableTaskRequest runnableTaskRequest) throws Exception {
+    //initialize start time for collecting latency metric
+    long startTime = System.currentTimeMillis();
     ByteBuffer requestBody = encodeTaskRequest(runnableTaskRequest);
 
-    return Retries.callWithRetries(() -> {
-      try {
-        HttpRequest.Builder requestBuilder = remoteClient.requestBuilder(HttpMethod.POST, TASK_WORKER_URL)
-          .withBody(requestBody.duplicate());
-        if (compression) {
-          requestBuilder.addHeader(HttpHeaders.CONTENT_ENCODING, "gzip");
-          requestBuilder.addHeader(HttpHeaders.ACCEPT_ENCODING, "gzip, deflate");
-        }
-        HttpRequest httpRequest = requestBuilder.build();
+    try {
+      RetryableTaskResult retryableTaskResult = Retries.callWithRetries((attempt) -> {
```

Comment:
The parameter should be `attempts` since it denotes the number of attempts, right?

Reply:
Done.